src/PostgREST/Query/SqlFragment.hs

```diff
@@ -266,24 +280,35 @@ pgFmtOrderTerm qi ot =
     maybe mempty nullOrder $ otNullOrder ot])
   where
     fmtOTerm = \case
-      OrderTerm{otTerm}                        -> pgFmtField qi otTerm
-      OrderRelationTerm{otRelation, otRelTerm} -> pgFmtField (QualifiedIdentifier mempty otRelation) otRelTerm
+      OrderTerm{otTerm=(fn, jp)}                        -> pgFmtField qi (unknownField fn jp)
+      OrderRelationTerm{otRelation, otRelTerm=(fn, jp)} -> pgFmtField (QualifiedIdentifier mempty otRelation) (unknownField fn jp)
 
     direction OrderAsc  = "ASC"
     direction OrderDesc = "DESC"
 
     nullOrder OrderNullsFirst = "NULLS FIRST"
     nullOrder OrderNullsLast  = "NULLS LAST"
 
-
-pgFmtFilter :: QualifiedIdentifier -> Filter -> SQL.Snippet
-pgFmtFilter _ (FilterNullEmbed hasNot fld) = SQL.sql (pgFmtIdent fld) <> " IS " <> (if hasNot then "NOT" else mempty) <> " NULL"
-pgFmtFilter _ (Filter _ (NoOpExpr _)) = mempty -- TODO unreachable because NoOpExpr is filtered on QueryParams
-pgFmtFilter table (Filter fld (OpExpr hasNot oper)) = notOp <> " " <> case oper of
+-- | Interpret a literal in the way the planner indicated through the CoercibleField.
+pgFmtUnknownLiteralForField :: SQL.Snippet -> CoercibleField -> SQL.Snippet
+pgFmtUnknownLiteralForField value CoercibleField{tfTransform=(Just parserProc)} = pgFmtCallUnary parserProc value
+-- But when no transform is requested, we just use the literal as-is.
+pgFmtUnknownLiteralForField value _ = value
+
+-- | Array version of the above, used by ANY().
+pgFmtArrayLiteralForField :: [Text] -> CoercibleField -> SQL.Snippet
+pgFmtArrayLiteralForField values CoercibleField{tfTransform=(Just parserProc)} = SQL.sql "ARRAY[" <> intercalateSnippet ", " (pgFmtCallUnary parserProc . unknownLiteral <$> values) <> "]"
```

Comment:
Q: The way this works indicates that the transformer function will be called `n` times for each element in the ARRAY right?

I've seen cases where the `in` values get really long, so this could be bad for perf.

Would it be possible to only have a single function call for this? Like defining a `CAST` from `text[]` to `domain`?

Reply:
Awesome, thanks. You're totally right, the way I did it is clumsy and could generate very verbose queries. We get something quite repetitive and long like,

`ANY (ARRAY[test.color('000100'), test.color('CAFE12'), test.color('01E240'), ...`

We have to call the parser on each value and I think that's what every `CAST` from `text[]` to `domain` would end up implementing, so might as well save the user the trouble and do that for them. 

I'm pushing an update that does it using a subquery and `unnest`. It's longer for small `n` with short parser function names but at the limits, if not more performant, at least a whole lot more debuggable and readable:

```
WITH pgrst_source AS (
  SELECT "test"."datarep_todos"."id", test.json("test"."datarep_todos"."due_at") AS "due_at"
  FROM "test"."datarep_todos"
  WHERE NOT "test"."datarep_todos"."label_color" = ANY (
    SELECT test.color(unnest('{000100,CAFE12,01E240}'::text[]))
  )
)
```

Reply:
Thanks! This looks much better.

I wanted to experiment with this code but somehow some tests are failing locally for me. I don't understand why because they're green on CI.

Pinging @laurenceisla to see if he can reproduce locally as well.

<details>
 <summary>postgrest-test-spec failures</summary>
<pre>
Failures:

  test/spec/Feature/Query/InsertSpec.hs:692:13:
  1) Feature.Query.InsertSpec, Data representations, on regular table, parses values in POST body and formats values in return=representation
       body mismatch:
         expected: [{"budget":"-100000000000000.13","created_at":-15,"due_at":"2018-01-03T11:00:00Z","icon_image":"3q2+7w==","id":5,"label_color":"#001100","name":"party"}]
         but got:  [{"id":5,"name":"party","label_color":"#001100","due_at":"2018-01-03T06:00:00-05:00","icon_image":"3q2+7w==","created_at":-15,"budget":"-100000000000000.13"}]

  To rerun use: --match "/Feature.Query.InsertSpec/Data representations/on regular table/parses values in POST body and formats values in return=representation/"

  test/spec/Feature/Query/InsertSpec.hs:703:13:
  2) Feature.Query.InsertSpec, Data representations, with ?columns parameter, ignores json keys not included in ?columns; parses only the ones specified
       body mismatch:
         expected: [{"due_at":"2018-01-01T00:00:00Z","id":5,"label_color":"#001100","name":null}]
         but got:  [{"id":5,"name":null,"label_color":"#001100","due_at":"2018-01-01T00:00:00-05:00"}]

  To rerun use: --match "/Feature.Query.InsertSpec/Data representations/with ?columns parameter/ignores json keys not included in ?columns; parses only the ones specified/"

  test/spec/Feature/Query/InsertSpec.hs:746:13:
  3) Feature.Query.InsertSpec, Data representations, on updatable view, parses values in POST body and formats values in return=representation
       body mismatch:
         expected: [{"dark_color":"#000880","due_at":"2018-01-03T11:00:00Z","id":5,"label_color":"#001100","name":"party"}]
         but got:  [{"id":5,"name":"party","label_color":"#001100","due_at":"2018-01-03T06:00:00-05:00","dark_color":"#000880"}]

  To rerun use: --match "/Feature.Query.InsertSpec/Data representations/on updatable view/parses values in POST body and formats values in return=representation/"

  test/spec/Feature/Query/InsertSpec.hs:757:13:
  4) Feature.Query.InsertSpec, Data representations, on updatable views with ?columns parameter, ignores json keys not included in ?columns; parses only the ones specified
       body mismatch:
         expected: [{"due_at":"2018-01-01T00:00:00Z","id":5,"label_color":"#001100","name":null}]
         but got:  [{"id":5,"name":null,"label_color":"#001100","due_at":"2018-01-01T00:00:00-05:00"}]

  To rerun use: --match "/Feature.Query.InsertSpec/Data representations/on updatable views with ?columns parameter/ignores json keys not included in ?columns; parses only the ones specified/"

  test/spec/Feature/Query/QuerySpec.hs:1286:65:
  5) Feature.Query.QuerySpec, Data representations for customisable value formatting and parsing, formats two columns with different formatters
       body mismatch:
         expected: "[{\"due_at\":\"2018-01-02T00:00:00Z\",\"id\":1,\"label_color\":\"#000000\"},{\"due_at\":\"2018-01-03T00:00:00Z\",\"id\":2,\"label_color\":\"#000100\"},{\"due_at\":\"2018-01
-01T14:12:34.123456Z\",\"id\":3,\"label_color\":\"#01E240\"}]"
         but got:  "[{\"id\":1,\"label_color\":\"#000000\",\"due_at\":\"2018-01-02T00:00:00-05:00\"}, \n {\"id\":2,\"label_color\":\"#000100\",\"due_at\":\"2018-01-03T00:00:00-05:00\"}, \n {\"
id\":3,\"label_color\":\"#01E240\",\"due_at\":\"2018-01-01T14:12:34.123456-05:00\"}]"

  To rerun use: --match "/Feature.Query.QuerySpec/Data representations for customisable value formatting and parsing/formats two columns with different formatters/"

  test/spec/Feature/Query/QuerySpec.hs:1314:73:
  6) Feature.Query.QuerySpec, Data representations for customisable value formatting and parsing, formats nulls
       body mismatch:
         expected: "[{\"due_at\":\"2018-01-01T14:12:34.123456Z\",\"id\":3,\"label_color\":\"#01E240\"},{\"due_at\":null,\"id\":4,\"label_color\":\"\"}]"
         but got:  "[{\"id\":3,\"label_color\":\"#01E240\",\"due_at\":\"2018-01-01T14:12:34.123456-05:00\"}, \n {\"id\":4,\"label_color\":\"\",\"due_at\":null}]"

  To rerun use: --match "/Feature.Query.QuerySpec/Data representations for customisable value formatting and parsing/formats nulls/"

  test/spec/Feature/Query/QuerySpec.hs:1318:45:
  7) Feature.Query.QuerySpec, Data representations for customisable value formatting and parsing, formats star select
       body mismatch:
         expected: "[{\"budget\":\"12.50\",\"created_at\":1513213350,\"due_at\":\"2018-01-02T00:00:00Z\",\"icon_image\":\"iVBORw0KGgoAAAANSUhEUgAAAAEAAAABAQAAAAA3bvkkAAAAABBJREFUeJxiYAEAAAAA//
8DAAAABgAFBXv6vUAAAAAASUVORK5CYII=\",\"id\":1,\"label_color\":\"#000000\",\"name\":\"Report\"},{\"budget\":\"100000000000000.13\",\"created_at\":1513213350,\"due_at\":\"2018-01-03T00:00:00Z\",
\"icon_image\":null,\"id\":2,\"label_color\":\"#000100\",\"name\":\"Essay\"},{\"budget\":\"0.00\",\"created_at\":1513213350,\"due_at\":\"2018-01-01T14:12:34.123456Z\",\"icon_image\":null,\"id\
":3,\"label_color\":\"#01E240\",\"name\":\"Algebra\"}]"
         but got:  "[{\"id\":1,\"name\":\"Report\",\"label_color\":\"#000000\",\"due_at\":\"2018-01-02T00:00:00-05:00\",\"icon_image\":\"iVBORw0KGgoAAAANSUhEUgAAAAEAAAABAQAAAAA3bvkkAAAAABBJREF
UeJxiYAEAAAAA//8DAAAABgAFBXv6vUAAAAAASUVORK5CYII=\",\"created_at\":1513231350,\"budget\":\"12.50\"}, \n {\"id\":2,\"name\":\"Essay\",\"label_color\":\"#000100\",\"due_at\":\"2018-01-03T00:00:0
0-05:00\",\"icon_image\":null,\"created_at\":1513231350,\"budget\":\"100000000000000.13\"}, \n {\"id\":3,\"name\":\"Algebra\",\"label_color\":\"#01E240\",\"due_at\":\"2018-01-01T14:12:34.12345
6-05:00\",\"icon_image\":null,\"created_at\":1513231350,\"budget\":\"0.00\"}]"

  To rerun use: --match "/Feature.Query.QuerySpec/Data representations for customisable value formatting and parsing/formats star select/"

  test/spec/Feature/Query/QuerySpec.hs:1326:36:
  8) Feature.Query.QuerySpec, Data representations for customisable value formatting and parsing, formats implicit star select
       body mismatch:
         expected: "[{\"budget\":\"12.50\",\"created_at\":1513213350,\"due_at\":\"2018-01-02T00:00:00Z\",\"icon_image\":\"iVBORw0KGgoAAAANSUhEUgAAAAEAAAABAQAAAAA3bvkkAAAAABBJREFUeJxiYAEAAAAA//
8DAAAABgAFBXv6vUAAAAAASUVORK5CYII=\",\"id\":1,\"label_color\":\"#000000\",\"name\":\"Report\"},{\"budget\":\"100000000000000.13\",\"created_at\":1513213350,\"due_at\":\"2018-01-03T00:00:00Z\",
\"icon_image\":null,\"id\":2,\"label_color\":\"#000100\",\"name\":\"Essay\"},{\"budget\":\"0.00\",\"created_at\":1513213350,\"due_at\":\"2018-01-01T14:12:34.123456Z\",\"icon_image\":null,\"id\
":3,\"label_color\":\"#01E240\",\"name\":\"Algebra\"}]"
         but got:  "[{\"id\":1,\"name\":\"Report\",\"label_color\":\"#000000\",\"due_at\":\"2018-01-02T00:00:00-05:00\",\"icon_image\":\"iVBORw0KGgoAAAANSUhEUgAAAAEAAAABAQAAAAA3bvkkAAAAABBJREF
UeJxiYAEAAAAA//8DAAAABgAFBXv6vUAAAAAASUVORK5CYII=\",\"created_at\":1513231350,\"budget\":\"12.50\"}, \n {\"id\":2,\"name\":\"Essay\",\"label_color\":\"#000100\",\"due_at\":\"2018-01-03T00:00:0
0-05:00\",\"icon_image\":null,\"created_at\":1513231350,\"budget\":\"100000000000000.13\"}, \n {\"id\":3,\"name\":\"Algebra\",\"label_color\":\"#01E240\",\"due_at\":\"2018-01-01T14:12:34.12345
6-05:00\",\"icon_image\":null,\"created_at\":1513231350,\"budget\":\"0.00\"}]"

  To rerun use: --match "/Feature.Query.QuerySpec/Data representations for customisable value formatting and parsing/formats implicit star select/"

  test/spec/Feature/Query/QuerySpec.hs:1334:52:
  9) Feature.Query.QuerySpec, Data representations for customisable value formatting and parsing, formats star and explicit mix
       body mismatch:
         expected: "[{\"budget\":\"12.50\",\"created_at\":1513213350,\"due_at\":\"2018-01-02T00:00:00Z\",\"icon_image\":\"iVBORw0KGgoAAAANSUhEUgAAAAEAAAABAQAAAAA3bvkkAAAAABBJREFUeJxiYAEAAAAA//
8DAAAABgAFBXv6vUAAAAAASUVORK5CYII=\",\"id\":1,\"label_color\":\"#000000\",\"name\":\"Report\"},{\"budget\":\"100000000000000.13\",\"created_at\":1513213350,\"due_at\":\"2018-01-03T00:00:00Z\",
\"icon_image\":null,\"id\":2,\"label_color\":\"#000100\",\"name\":\"Essay\"},{\"budget\":\"0.00\",\"created_at\":1513213350,\"due_at\":\"2018-01-01T14:12:34.123456Z\",\"icon_image\":null,\"id\
":3,\"label_color\":\"#01E240\",\"name\":\"Algebra\"}]"
         but got:  "[{\"due_at\":\"2018-01-02T00:00:00-05:00\",\"id\":1,\"name\":\"Report\",\"label_color\":\"#000000\",\"due_at\":\"2018-01-02T00:00:00-05:00\",\"icon_image\":\"iVBORw0KGgoAAA
ANSUhEUgAAAAEAAAABAQAAAAA3bvkkAAAAABBJREFUeJxiYAEAAAAA//8DAAAABgAFBXv6vUAAAAAASUVORK5CYII=\",\"created_at\":1513231350,\"budget\":\"12.50\"}, \n {\"due_at\":\"2018-01-03T00:00:00-05:00\",\"id\
":2,\"name\":\"Essay\",\"label_color\":\"#000100\",\"due_at\":\"2018-01-03T00:00:00-05:00\",\"icon_image\":null,\"created_at\":1513231350,\"budget\":\"100000000000000.13\"}, \n {\"due_at\":\"2
018-01-01T14:12:34.123456-05:00\",\"id\":3,\"name\":\"Algebra\",\"label_color\":\"#01E240\",\"due_at\":\"2018-01-01T14:12:34.123456-05:00\",\"icon_image\":null,\"created_at\":1513231350,\"budg
et\":\"0.00\"}]"

  To rerun use: --match "/Feature.Query.QuerySpec/Data representations for customisable value formatting and parsing/formats star and explicit mix/"

  test/spec/Feature/Query/QuerySpec.hs:1342:139:
  10) Feature.Query.QuerySpec, Data representations for customisable value formatting and parsing, formats through join
       body mismatch:
         expected: "[{\"first_item\":{\"due_at\":\"2018-01-03T00:00:00Z\",\"label_color\":\"#000100\"},\"id\":1,\"name\":\"school related\"},{\"first_item\":{\"due_at\":\"2018-01-02T00:00:00Z\
",\"label_color\":\"#000000\"},\"id\":2,\"name\":\"do these first\"}]"
         but got:  "[{\"id\":1,\"name\":\"school related\",\"first_item\":{\"label_color\":\"#000100\",\"due_at\":\"2018-01-03T00:00:00-05:00\"}}, \n {\"id\":2,\"name\":\"do these first\",\"fi
rst_item\":{\"label_color\":\"#000000\",\"due_at\":\"2018-01-02T00:00:00-05:00\"}}]"

  To rerun use: --match "/Feature.Query.QuerySpec/Data representations for customisable value formatting and parsing/formats through join/"

  test/spec/Feature/Query/QuerySpec.hs:1346:124:
  11) Feature.Query.QuerySpec, Data representations for customisable value formatting and parsing, formats through join with star select
       body mismatch:
         expected: "[{\"id\":1,\"name\":\"school related\",\"second_item\":{\"budget\":\"0.00\",\"created_at\":1513213350,\"due_at\":\"2018-01-01T14:12:34.123456Z\",\"icon_image\":null,\"id\":
3,\"label_color\":\"#01E240\",\"name\":\"Algebra\"}},{\"id\":2,\"name\":\"do these first\",\"second_item\":{\"budget\":\"0.00\",\"created_at\":1513213350,\"due_at\":\"2018-01-01T14:12:34.12345
6Z\",\"icon_image\":null,\"id\":3,\"label_color\":\"#01E240\",\"name\":\"Algebra\"}}]"
         but got:  "[{\"id\":1,\"name\":\"school related\",\"second_item\":{\"id\":3,\"name\":\"Algebra\",\"label_color\":\"#01E240\",\"due_at\":\"2018-01-01T14:12:34.123456-05:00\",\"icon_ima
ge\":null,\"created_at\":1513231350,\"budget\":\"0.00\"}}, \n {\"id\":2,\"name\":\"do these first\",\"second_item\":{\"id\":3,\"name\":\"Algebra\",\"label_color\":\"#01E240\",\"due_at\":\"2018
-01-01T14:12:34.123456-05:00\",\"icon_image\":null,\"created_at\":1513231350,\"budget\":\"0.00\"}}]"

  To rerun use: --match "/Feature.Query.QuerySpec/Data representations for customisable value formatting and parsing/formats through join with star select/"

  test/spec/Feature/Query/QuerySpec.hs:1353:67:
  12) Feature.Query.QuerySpec, Data representations for customisable value formatting and parsing, uses text parser on value for filter given through query parameters
       body mismatch:
         expected: [{"due_at":"2018-01-03T00:00:00Z","id":2}]
         but got:  [{"id":2,"due_at":"2018-01-03T00:00:00-05:00"}]

  To rerun use: --match "/Feature.Query.QuerySpec/Data representations for customisable value formatting and parsing/uses text parser on value for filter given through query parameters/"

  test/spec/Feature/Query/QuerySpec.hs:1367:78:
  13) Feature.Query.QuerySpec, Data representations for customisable value formatting and parsing, uses text parser for filter with 'IN' predicates
       status mismatch:
         expected: 200
         but got:  400
       body mismatch:
         expected: [{"due_at":"2018-01-03T00:00:00Z","id":2},{"due_at":"2018-01-01T14:12:34.123456Z","id":3}]
         but got:  {"code":"42601","details":null,"hint":null,"message":"syntax error at or near \"SELECT\""}

  To rerun use: --match "/Feature.Query.QuerySpec/Data representations for customisable value formatting and parsing/uses text parser for filter with 'IN' predicates/"

  test/spec/Feature/Query/QuerySpec.hs:1374:82:
  14) Feature.Query.QuerySpec, Data representations for customisable value formatting and parsing, uses text parser for filter with 'NOT IN' predicates
       status mismatch:
         expected: 200
         but got:  400
       body mismatch:
         expected: [{"due_at":"2018-01-03T00:00:00Z","id":2}]
         but got:  {"code":"42601","details":null,"hint":null,"message":"syntax error at or near \"SELECT\""}

  To rerun use: --match "/Feature.Query.QuerySpec/Data representations for customisable value formatting and parsing/uses text parser for filter with 'NOT IN' predicates/"

  test/spec/Feature/Query/QuerySpec.hs:1380:165:
  15) Feature.Query.QuerySpec, Data representations for customisable value formatting and parsing, uses text parser on value for filter across relations
       body mismatch:
         expected: "[{\"datarep_todos\":null,\"id\":1,\"name\":\"school related\"},{\"datarep_todos\":{\"due_at\":\"2018-01-02T00:00:00Z\",\"label_color\":\"#000000\"},\"id\":2,\"name\":\"do t
hese first\"}]"
         but got:  "[{\"id\":1,\"name\":\"school related\",\"datarep_todos\":null}, \n {\"id\":2,\"name\":\"do these first\",\"datarep_todos\":{\"label_color\":\"#000000\",\"due_at\":\"2018-01
-02T00:00:00-05:00\"}}]"

  To rerun use: --match "/Feature.Query.QuerySpec/Data representations for customisable value formatting and parsing/uses text parser on value for filter across relations/"

  test/spec/Feature/Query/UpdateSpec.hs:577:13:
  16) Feature.Query.UpdateSpec, Data representations, for a single row, parses values in payload and formats values in return=representation
       body mismatch:
         expected: [{"budget":"100000000000000.13","created_at":1513213350,"due_at":"2019-01-03T11:00:20Z","icon_image":"3q2+7w==","id":2,"label_color":"#221100","name":"Essay"}]
         but got:  [{"id":2,"name":"Essay","label_color":"#221100","due_at":"2019-01-03T06:00:20-05:00","icon_image":"3q2+7w==","created_at":1513231350,"budget":"100000000000000.13"}]

  To rerun use: --match "/Feature.Query.UpdateSpec/Data representations/for a single row/parses values in payload and formats values in return=representation/"

  test/spec/Feature/Query/UpdateSpec.hs:587:13:
  17) Feature.Query.UpdateSpec, Data representations, for a single row, parses values in payload and formats star mixed selected values in return=representation
       body mismatch:
         expected: [{"budget":"100000000000000.13","created_at":0,"due_at":"2019-01-03T11:00:00Z","icon_image":null,"id":2,"label_color":"#221100","name":"Essay"}]
         but got:  [{"due_at":"2019-01-03T06:00:00-05:00","id":2,"name":"Essay","label_color":"#221100","due_at":"2019-01-03T06:00:00-05:00","icon_image":null,"created_at":0,"budget":"10000000
0000000.13"}]

  To rerun use: --match "/Feature.Query.UpdateSpec/Data representations/for a single row/parses values in payload and formats star mixed selected values in return=representation/"

  test/spec/Feature/Query/UpdateSpec.hs:612:13:
  18) Feature.Query.UpdateSpec, Data representations, for multiple rows, parses values in payload and formats values in return=representation
       body mismatch:
         expected: "[{\"budget\":\"12.50\",\"created_at\":1513213350,\"due_at\":\"2019-01-03T11:00:00Z\",\"icon_image\":\"3q2+7w==\",\"id\":1,\"label_color\":\"#221100\",\"name\":\"Report\"},{
\"budget\":\"100000000000000.13\",\"created_at\":1513213350,\"due_at\":\"2019-01-03T11:00:00Z\",\"icon_image\":\"3q2+7w==\",\"id\":2,\"label_color\":\"#221100\",\"name\":\"Essay\"},{\"budget\"
:\"0.00\",\"created_at\":1513213350,\"due_at\":\"2019-01-03T11:00:00Z\",\"icon_image\":\"3q2+7w==\",\"id\":3,\"label_color\":\"#221100\",\"name\":\"Algebra\"}]"
         but got:  "[{\"id\":1,\"name\":\"Report\",\"label_color\":\"#221100\",\"due_at\":\"2019-01-03T06:00:00-05:00\",\"icon_image\":\"3q2+7w==\",\"created_at\":1513231350,\"budget\":\"12.50
\"}, \n {\"id\":2,\"name\":\"Essay\",\"label_color\":\"#221100\",\"due_at\":\"2019-01-03T06:00:00-05:00\",\"icon_image\":\"3q2+7w==\",\"created_at\":1513231350,\"budget\":\"100000000000000.13\
"}, \n {\"id\":3,\"name\":\"Algebra\",\"label_color\":\"#221100\",\"due_at\":\"2019-01-03T06:00:00-05:00\",\"icon_image\":\"3q2+7w==\",\"created_at\":1513231350,\"budget\":\"0.00\"}]"

  To rerun use: --match "/Feature.Query.UpdateSpec/Data representations/for multiple rows/parses values in payload and formats values in return=representation/"

  test/spec/Feature/Query/UpdateSpec.hs:626:13:
  19) Feature.Query.UpdateSpec, Data representations, with ?columns parameter, ignores json keys not included in ?columns; parses only the ones specified
       body mismatch:
         expected: [{"budget":"100000000000000.13","created_at":1513213350,"due_at":"2019-01-03T11:00:00Z","icon_image":null,"id":2,"label_color":"#000100","name":"Essay"}]
         but got:  [{"id":2,"name":"Essay","label_color":"#000100","due_at":"2019-01-03T06:00:00-05:00","icon_image":null,"created_at":1513231350,"budget":"100000000000000.13"}]

  To rerun use: --match "/Feature.Query.UpdateSpec/Data representations/with ?columns parameter/ignores json keys not included in ?columns; parses only the ones specified/"

  test/spec/Feature/Query/UpdateSpec.hs:673:15:
  20) Feature.Query.UpdateSpec, Data representations, on a view, for a single row, parses values in payload and formats values in return=representation
       body mismatch:
         expected: [{"dark_color":"#110880","due_at":"2019-01-03T11:00:20Z","id":2,"label_color":"#221100","name":"Essay"}]
         but got:  [{"id":2,"name":"Essay","label_color":"#221100","due_at":"2019-01-03T06:00:20-05:00","dark_color":"#110880"}]

  To rerun use: --match "/Feature.Query.UpdateSpec/Data representations/on a view/for a single row/parses values in payload and formats values in return=representation/"

  test/spec/Feature/Query/UpdateSpec.hs:697:15:
  21) Feature.Query.UpdateSpec, Data representations, on a view, for multiple rows, parses values in payload and formats values in return=representation
       body mismatch:
         expected: "[{\"dark_color\":\"#110880\",\"due_at\":\"2019-01-03T11:00:00Z\",\"id\":1,\"label_color\":\"#221100\",\"name\":\"Report\"},{\"dark_color\":\"#110880\",\"due_at\":\"2019-01-
03T11:00:00Z\",\"id\":2,\"label_color\":\"#221100\",\"name\":\"Essay\"},{\"dark_color\":\"#110880\",\"due_at\":\"2019-01-03T11:00:00Z\",\"id\":3,\"label_color\":\"#221100\",\"name\":\"Algebra\
"}]"
         but got:  "[{\"id\":1,\"name\":\"Report\",\"label_color\":\"#221100\",\"due_at\":\"2019-01-03T06:00:00-05:00\",\"dark_color\":\"#110880\"}, \n {\"id\":2,\"name\":\"Essay\",\"label_col
or\":\"#221100\",\"due_at\":\"2019-01-03T06:00:00-05:00\",\"dark_color\":\"#110880\"}, \n {\"id\":3,\"name\":\"Algebra\",\"label_color\":\"#221100\",\"due_at\":\"2019-01-03T06:00:00-05:00\",\"
dark_color\":\"#110880\"}]"

  To rerun use: --match "/Feature.Query.UpdateSpec/Data representations/on a view/for multiple rows/parses values in payload and formats values in return=representation/"

  test/spec/Feature/Query/UpdateSpec.hs:711:15:
  22) Feature.Query.UpdateSpec, Data representations, on a view, with ?columns parameter, ignores json keys not included in ?columns; parses only the ones specified
       body mismatch:
         expected: [{"dark_color":"#000080","due_at":"2019-01-03T11:00:00Z","id":2,"label_color":"#000100","name":"Essay"}]
         but got:  [{"id":2,"name":"Essay","label_color":"#000100","due_at":"2019-01-03T06:00:00-05:00","dark_color":"#000080"}]

  To rerun use: --match "/Feature.Query.UpdateSpec/Data representations/on a view/with ?columns parameter/ignores json keys not included in ?columns; parses only the ones specified/"

Randomized with seed 709440945

Finished in 19.5713 seconds
1045 examples, 22 failures, 1 pending
</pre>
</details>

Reply:
Are you sure you're setting `PGTZ=UTC` when running your local Postgres for testing? All the `postgrest-with-postgresql-X` nix tools should do this by default and that's why the tests work on CI. Since we're testing with date and time strings here, the tests make some assumptions about what TZ we're formatting for.

Reply:
Thanks! That was it.

>  All the postgrest-with-postgresql-X nix tools should do this by default and that's why the tests work on CI

Nope, they don't. Only when doing `PGTZ=UTC postgrest-test-spec` explicitly they work.

Reply:
Fixed on https://github.com/PostgREST/postgrest/pull/2666